python/ray/data/dataset.py

```diff
@@ -2792,6 +2792,7 @@ def iter_batches(
         batch_size: Optional[int] = 256,
         batch_format: str = "default",
         drop_last: bool = False,
+        batch_fn: Optional[Callable[[DataBatch], DataBatch]] = None,
```

Comment:
why don't we always call them collate_fn? since the intention is to give people things that they are familiar with.

Reply:
The reasoning is that collate_fn is familiar term from torch DataLoader…but wasn’t sure it made sense for generic iter_batches

Reply:
I think we should also just make it collate_fn here, i think consistency makes sense (especially among similar apis)

Reply:
agree

Reply:
updated